File: docs/vs-2015/snippets/visualbasic/VS_Snippets_VSSDK/creatingalanguageservice(managedpackageframework)/vb/mylanguageservice.vb
'Copyright (c) Microsoft Corporation. All rights reserved.
'<Snippet1>
Imports Microsoft.VisualStudio
Imports Microsoft.VisualStudio.Package
Imports Microsoft.VisualStudio.TextManager.Interop
Imports Microsoft.VisualStudio.OLE.Interop
'</Snippet1>

Namespace Microsoft.MyLanguagePackage
    '<Snippet2>
	Friend Class MyLanguageService
		Inherits LanguageService
        '</Snippet2>

		Public Overrides Function GetFormatFilterList() As String
			Throw New NotImplementedException()
		End Function

		Public Overrides Function GetLanguagePreferences() As LanguagePreferences
			Throw New NotImplementedException()
		End Function

		Public Overrides Function GetScanner(ByVal buffer As IVsTextLines) As IScanner
			Throw New NotImplementedException()
		End Function

		Public Overrides ReadOnly Property Name() As String
			Get
				Throw New NotImplementedException()
			End Get
		End Property

		Public Overrides Function ParseSource(ByVal req As ParseRequest) As AuthoringScope
			Throw New NotImplementedException()
		End Function
	End Class
End Namespace

File: docs/vs-2015/snippets/csharp/VS_Snippets_VSSDK/creatingalanguageservice(managedpackageframework)/cs/mylanguageservice.cs
//<Snippet1>
using Microsoft.VisualStudio;
using Microsoft.VisualStudio.Package;
using Microsoft.VisualStudio.TextManager.Interop;
using Microsoft.VisualStudio.OLE.Interop;
//</Snippet1>

namespace Microsoft.MyLanguagePackage
{
   //<Snippet2>
    class MyLanguageService : LanguageService
   //</Snippet2>
    {
        public override string GetFormatFilterList()
        {
            throw new System.NotImplementedException();
        }

        public override LanguagePreferences GetLanguagePreferences()
        {
            throw new System.NotImplementedException();
        }

        public override IScanner GetScanner(IVsTextLines buffer)
        {
            throw new System.NotImplementedException();
        }

        public override string Name
        {
            get { throw new System.NotImplementedException(); }
        }

        public override AuthoringScope ParseSource(ParseRequest req)
        {
            throw new System.NotImplementedException();
        }
    }
}
